cirq/ops/parity_gates.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Quantum gates that phase with respect to product-of-pauli observables."""
+
+from typing import Union, Sequence
+
+import numpy as np
+
+from cirq import protocols, linalg
+from cirq.ops import gate_features, eigen_gate
+from cirq.ops.common_gates import _rads_func_symbol
+from cirq.type_workarounds import NotImplementedType
+
+
+class XXPowGate(eigen_gate.EigenGate,
```

Comment:
Do you plan to add `_apply_unitary_to_tensor_()` for XX and YY in a follow-up PR? It's going to be a little surprising that ZZ has it, but the other two gates don't.

Reply:
No plans for that, because I expect it would be slower than the default `einsum` based method. ZZ is done special because it is diagonal. XX and YY aren't particularly sparse, except at exponent=1 which I don't think happens often.